### DISCUSSION TITLE
Implementing Deployment Preview GHA

### DISCUSSION THREAD

**Discussion by Sukriti-sood** [open-horizon/open-horizon.github.io#198] [Ideas]
The purpose of this discussion is to capture all ways in which we can implement Deployment Preview GHA.

One of way I found by taking inspiration from [here](https://github.com/blueedgetechno/win11React/blob/master/.github/workflows/gen-preview.yml) is we can have **Open-Horizon-Bot**  Account that will create repostory of pr opened and deploy it on gh-pages and we will have preview link of that PR in comment, and Once the PR is merged, we can delete the repo from **Open-Horizon-Bot** Account.

**Comment by joewxboy** [comment]
> we can ... create repository of PR opened and deploy it on gh-pages and we will have preview link of that PR in comment, and Once the PR is merged, we can delete the repo from **Open-Horizon-Bot** Account.

Seems like this approach will only allow you to preview a single PR at a time, right?

And is the only reason this approach is being used is that it builds and tests the site with github-pages so it's a "perfect" preview for what will happen in production?

**Reply by Sukriti-sood** [reply]
GHA will run on event when PR is opened so whenever pr is opened or updated ,it will display the preview.

**Reply by joewxboy** [reply]
What I mean is ... it appears that it is hard-coded to a specific single repository, so only one PR can be tested.  If that PR is still in the process of being tested another PR will have to wait until this one is merged or closed, right?

**Reply by Sukriti-sood** [reply]
Its not hard coded to a specific single repository... There is a command `gh repo create win11bot/pr-${{ github.event.pull_request.number }} --public` which is creating repository with PR number.

**Comment by joewxboy** [comment]
Thanks, @Sukriti-sood .  What is a good way to test this?

**Reply by Sukriti-sood** [reply]
@joewxboy If you meant demo, you can Check [this](https://github.com/blueedgetechno/win11React/pull/319) PR.

**Reply by joewxboy** [reply]
OK, that looks really nice!  You've answered all my questions.

**Comment by Abhijay007** [comment]
@johnwalicki do you have any suggestions for deployment Preview bot? @Sukriti-sood have some good suggestions, so john can you please tell us whether we are following the right approach or not?

**Comment by joewxboy** [comment]
BTW, this approach may also help developers editing repo-based docs (like anax repo, /docs folder) to preview their documentation and resolve variables, check hyperlinks, and confirm look-and-feel.
